src/gradient.rs

```diff
@@ -56,131 +72,132 @@ impl From<(f32, Color)> for ColorStop {
 /// Collection of color stops.
 pub type ColorStops = SmallVec<[ColorStop; 4]>;
 
-/// Definition of a gradient that transitions between two or more colors along
-/// a line.
-#[derive(Clone, PartialEq, Default, Debug)]
-pub struct LinearGradient {
-    pub start: Point,
-    pub end: Point,
+/// Properties for the supported gradient types.
+#[derive(Clone, PartialEq, Debug)]
+pub enum GradientKind {
```

Comment:
The word "kind" seems slightly wrong, as I'd expect it to be a simple enum. But I can't immediately think of a better word, and that will be clear from usage anyway.

Reply:
I don't disagree, but the pattern of having common fields in `S` and variant dependent fields `SKind` pops up all over the rustc codebase so I think I internalized it?